internal/impl/redpanda/processor_data_transforms.go

```diff
@@ -0,0 +1,414 @@
+		inputKey, err := conf.FieldString("output_key")
+		if err != nil {
+			return nil, err
+		}
+		cfg.outputKeyField = &inputKey
+	}
+
+	if conf.Contains("input_headers") {
+		inputMetadata, err := conf.FieldMetadataFilter("input_headers")
+		if err != nil {
+			return nil, err
+		}
+		cfg.inputMetadata = inputMetadata
+	}
+
+	if conf.Contains("output_metadata") {
+		outputMetadata, err := conf.FieldMetadataFilter("output_metadata")
+		if err != nil {
+			return nil, err
+		}
+		cfg.outputMetadata = outputMetadata
+	}
+
+	if conf.Contains("timestamp") {
+		ts, err := conf.FieldInterpolatedString("timestamp")
+		if err != nil {
+			return nil, err
+		}
+		cfg.timestamp = ts
+	}
+
+	timeout, err := conf.FieldDuration("timeout")
+	if err != nil {
+		return nil, err
+	}
+	cfg.timeout = timeout
+
+	maxMemoryPages, err := conf.FieldInt("max_memory_pages")
+	if err != nil {
+		return nil, err
+	}
+	cfg.maxMemoryPages = maxMemoryPages
+
+	return newDataTransformProcessor(fileBytes, cfg, mgr)
+}
+
+func newDataTransformProcessor(wasmBinary []byte, cfg dataTransformConfig, mgr *service.Resources) (*dataTransformEnginePool, error) {
+	ctx := context.Background()
+	runtimeCfg := wazero.NewRuntimeConfig().
+		WithCloseOnContextDone(true).
+		WithCompilationCache(wazero.NewCompilationCache()).
+		WithMemoryLimitPages(uint32(cfg.maxMemoryPages))
+	r := wazero.NewRuntimeWithConfig(ctx, runtimeCfg)
+	cm, err := r.CompileModule(ctx, wasmBinary)
+	_ = r.Close(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Validate ABI contract
+	proc := &dataTransformEnginePool{
+		log:           mgr.Logger(),
+		modulePool:    sync.Pool{},
+		runtimeConfig: runtimeCfg,
+		wasmBinary:    cm,
+		cfg:           cfg,
+	}
+	// Ensure we can create at least one module runner.
+	modRunner, err := proc.newModule()
+	if err != nil {
+		return nil, err
+	}
+
+	proc.modulePool.Put(modRunner)
+	return proc, nil
+}
+
+func (p *dataTransformEnginePool) newModule() (engine *dataTransformEngine, err error) {
+	ctx := context.Background()
+	r := wazero.NewRuntimeWithConfig(ctx, p.runtimeConfig)
+	engine = &dataTransformEngine{
+		log:       p.log,
+		cfg:       &p.cfg,
+		runtime:   r,
+		hostChan:  make(chan any),
+		guestChan: make(chan any),
+		procErr:   nil,
+	}
+	defer func() {
+		if err != nil {
+			engine.runtime.Close(context.Background())
+		}
+	}()
+
+	builder := r.NewHostModuleBuilder("redpanda_transform")
+	for name, ctor := range transformHostFunctions {
+		builder = builder.NewFunctionBuilder().WithFunc(ctor(engine)).Export(name)
+	}
+	if _, err = builder.Instantiate(ctx); err != nil {
+		return
+	}
+
+	if _, err = wasi_snapshot_preview1.Instantiate(ctx, r); err != nil {
+		return
+	}
+	cfg := wazero.NewModuleConfig().
+		WithStartFunctions().
+		WithArgs("transform").
+		WithName("transform").
+		WithEnv("REDPANDA_INPUT_TOPIC", "benthos")
+	for i := 0; i < 8; i += 1 {
+		cfg = cfg.WithEnv(fmt.Sprintf("REDPANDA_OUTPUT_TOPIC_%d", i), fmt.Sprintf("output_%d", i))
```

Comment:
I just created dummy values here - I'm not sure if there should be anything special